src/sagemaker/jumpstart/curated_hub/utils.py

```diff
@@ -164,3 +179,97 @@ def create_hub_bucket_if_it_does_not_exist(
+        ]
+      ))
+
+    return responses
+    
+def find_jumpstart_tags_for_model(hub_name: str, hub_content_name: str, region: str, session: Session) -> List[Tag]:
+    list_versions_response = session.list_hub_content_versions(
+        hub_name=hub_name,
+        hub_content_type='Model',
+        hub_content_name=hub_content_name
+    )
+    hub_content_versions = list_versions_response["HubContentSummaries"]
+
+    tag_name_to_versions_map: Dict[CuratedHubTagName, List[str]] = {}
+    for hub_content_version_summary in hub_content_versions:
+        jumpstart_model = get_jumpstart_model_and_version(hub_content_version_summary)
+        if jumpstart_model["model_id"] is None or jumpstart_model["version"] is None:
+            continue
+        tag_names_to_add: List[CuratedHubTagName] = find_jumpstart_tags_for_model_version(
+            model_id=jumpstart_model["model_id"],
+            version=jumpstart_model["version"],
+            region=region,
+            session=session
+        )
+
+        for tag_name in tag_names_to_add:
+          if tag_name not in tag_name_to_versions_map:
+              tag_name_to_versions_map[tag_name] = []
+          tag_name_to_versions_map[tag_name].append(hub_content_version_summary["HubContentVersion"])
+      
+    tags: List[Tag] = []
+    for tag_name, versions in tag_name_to_versions_map.items():
+        tags.append(Tag(
+            key=tag_name,
+            value=str(versions)
+        ))
+
+    return tags
+
+    
+    
+def find_jumpstart_tags_for_model_version(model_id: str, version: str, region: str, session: Session) -> List[CuratedHubTagName]:
+    tags_to_add: List[CuratedHubTagName] = []
+    specs = utils.verify_model_region_and_return_specs(
+        model_id=model_id, 
+        version=version,
+        region=region,
+        scope=JumpStartScriptScope.INFERENCE,
+        tolerate_vulnerable_model = True,
+        tolerate_deprecated_model = True,
+        sagemaker_session=session,
+    )
+
+    if (specs.deprecated):
+          tags_to_add.append(CuratedHubTagName.DEPRECATED_VERSIONS_TAG)
+    if (specs.inference_vulnerable):
+        tags_to_add.append(CuratedHubTagName.INFERENCE_VULNERABLE_VERSIONS_TAG)
+    if (specs.training_vulnerable):
+        tags_to_add.append(CuratedHubTagName.TRAINING_VULNERABLE_VERSIONS_TAG)
+
+    return tags_to_add
+
+    
+
+def get_jumpstart_model_and_version(hub_content_summary: Dict[str, Any]) -> Dict[str, Any]:
+    jumpstart_model_id = next(
+        (
+            tag
+            for tag in hub_content_summary["search_keywords"]
+            if tag.startswith(JUMPSTART_HUB_MODEL_ID_TAG_PREFIX)
+        ),
+        None,
+    )
+    jumpstart_model_version = next(
+        (
+            tag
+            for tag in hub_content_summary["search_keywords"]
+            if tag.startswith(JUMPSTART_HUB_MODEL_VERSION_TAG_PREFIX)
+        ),
+        None,
+    )
+
+    return {"model_id": jumpstart_model_id, "version": jumpstart_model_version}
```

Comment:
To make this function easier to use, can we either return `None` or the object? Then, you can update the logic on line 209 to be `if not jumpstart_model:`. We never want to handle a partial response where we either have the model id or version- it's all or nothing

Reply:
Sure, we can do that. I preserved the logic from previous to not break the rest of the code already written